src/main/kotlin/screepsai/roles/Harvester.kt

```diff
@@ -1,21 +1,64 @@
 package screepsai.roles
 
 import screeps.api.*
+import screeps.utils.memory.memory
+
+var CreepMemory.energySource: String? by memory()
+var RoomMemory.energySourceAssignments: Array<String?> by memory { arrayOf(null, null) }
```

Comment:
This would probably be better as a map of energy source ID to creep id.

With javascript this would be easy, need to figure out how to interact with memory with arbitrary keys using the kotlin interface though.

Reply:
Fiddled with this for a while and ended up completely mangling the memory and causing errors long enough for all the living creeps to expire.

May want to explore arbitrary memory structures like this later as it's own project. There don't seem to be very many docs for how to use the kotlin screeps memory module.